File: DesktopSample/Advanced/Section/CustomDrillThrough/C#/CustomDrillThrough/ViewerForm.cs
using GrapeCity.ActiveReports;
using System;
using System.Text;
using System.Windows.Forms;
using System.Xml;

namespace ActiveReports.Samples.CustomDrillThrough
{
	/// <summary>
	///frmviewmainの概要の説明です。 
	/// 
	///本サンプルはハイパーリンクおよびビューワのハイパーリンクイベントを使用し、 
	///ドリルダウンレポートをシミュレートする方法を表します。
	/// </summary>
	public partial class ViewerForm : System.Windows.Forms.Form
	{
		/// <summary>
		/// 必要なデザイナ変数です。
		/// </summary>
		private readonly bool _loadMainReport;

		public ViewerForm()
		{
			//
			// Windows フォーム デザイナ サポートに必要です。
			//
			InitializeComponent();
			_loadMainReport = true;
		}

		public ViewerForm(bool loadMainReport)
		{
			InitializeComponent();
			_loadMainReport = loadMainReport;
		}

		/// <summary>
		/// 
		/// アプリケーションのメインエントリポイントです。
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
#if NET6_0_OR_GREATER
			Application.SetHighDpiMode(HighDpiMode.DpiUnawareGdiScaled);
#endif
            Application.EnableVisualStyles();
            Application.SetCompatibleTextRenderingDefault(false);
            Application.Run(new ViewerForm());
		}

		/// <summary>
		/// arvMain_HyperLink: ビューワオブジェクトから送信されたハイパーリンクを処理します。
		/// </summary>
		private void arvMain_HyperLink(object sender, GrapeCity.ActiveReports.Viewer.Win.HyperLinkEventArgs e)
		{
			e.Handled = true;
			string hyperlink = e.HyperLink.Split(':')[1];
			string report = e.HyperLink.Split(':')[0];

			if (report == "DrillThrough1")
			{
				// 顧客番号をクリックして、ドリルダウンレポートに移動し、顧客情報を表示します。
				var rpt2 = new SectionReport();
				rpt2.LoadLayout(XmlReader.Create(Properties.Resources.DrillThrough1));
				ViewerForm frm2 = new ViewerForm(false);
				rpt2.Parameters["customerID"].Value = hyperlink;
				frm2.arvMain.LoadDocument(rpt2);
				frm2.ShowDialog(this);
			}
			else if (report == "DrillThrough2")
			{
				// 受注番号をクリックし、受注情報を開きます。
				var rpt3 = new SectionReport();
				rpt3.LoadLayout(XmlReader.Create(Properties.Resources.DrillThrough2));
				ViewerForm frm3 = new ViewerForm(false);
				rpt3.Parameters["orderID"].Value = hyperlink;
				frm3.arvMain.LoadDocument(rpt3);
				frm3.ShowDialog(this);
			}
		}

		/// <summary>
		/// ViewerForm_Load - フォームで初期設定を行います。
		/// </summary>
		private void ViewerForm_Load(object sender, EventArgs e)
		{
			if(_loadMainReport)
			{
				var rpt = new SectionReport();
				rpt.LoadLayout(XmlReader.Create(Properties.Resources.DrillThroughMain));
				arvMain.LoadDocument(rpt);
			}
		}
	}
}

File: DesktopSample/Advanced/Section/CustomDrillThrough/VB.NET/CustomDrillThrough/ViewerForm.vb
Imports System
Imports System.Text
Imports System.Windows.Forms
Imports System.Xml
Imports GrapeCity.ActiveReports

Public Class ViewerForm
	Inherits System.Windows.Forms.Form

	Private _loadMainReport As Boolean

	Public Sub New()
		MyBase.New()
		InitializeComponent()
		_loadMainReport = True
	End Sub

	Public Sub New(ByVal loadMainReport As Boolean)
		MyBase.New()
		InitializeComponent()
		_loadMainReport = loadMainReport
	End Sub

	Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
		If disposing Then
			If Not (components Is Nothing) Then
				components.Dispose()
			End If
		End If
		MyBase.Dispose(disposing)
	End Sub

	<STAThread()>
	Shared Sub Main()
		Encoding.RegisterProvider(CodePagesEncodingProvider.Instance)
		Application.Run(New ViewerForm())
	End Sub 'Main

	' arvMain_HyperLink: ビューワオブジェクトから送信されたハイパーリンクを処理します。
	Private Sub arvMain_HyperLink(ByVal sender As Object, ByVal e As GrapeCity.ActiveReports.Viewer.Win.HyperLinkEventArgs) Handles arvMain.HyperLink
		e.Handled = True
		Dim hyperlink As String = e.HyperLink.Split(New String() {":"}, StringSplitOptions.None)(1)
		Dim report As String = e.HyperLink.Split(New String() {":"}, StringSplitOptions.None)(0)
		If report = "DrillThrough1" Then
			' 顧客番号をクリックして、ドリルダウンレポートに移動し、顧客情報を表示します。
			Dim rpt2 As New SectionReport
			rpt2.LoadLayout(XmlReader.Create(My.Resources.DrillThrough1))
			Dim frm2 As New ViewerForm(False)
			rpt2.Parameters("customerID").Value = hyperlink
			frm2.arvMain.LoadDocument(rpt2)
			frm2.ShowDialog(Me)
		ElseIf report = "DrillThrough2" Then
			' 受注番号をクリックして受注情報を開きます。
			Dim rpt3 As New SectionReport
			rpt3.LoadLayout(XmlReader.Create(My.Resources.DrillThrough2))
			Dim frm3 As New ViewerForm(False)
			rpt3.Parameters("orderID").Value = hyperlink
			frm3.arvMain.LoadDocument(rpt3)
			frm3.ShowDialog(Me)
		End If
	End Sub 'arvMain_HyperLink

	' ViewerForm_Load - フォームで初期設定を行います。
	Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
		If _loadMainReport Then
			Dim rpt As New SectionReport
			rpt.LoadLayout(XmlReader.Create(My.Resources.DrillThroughMain))
			arvMain.LoadDocument(rpt)
		End If
	End Sub 'Form1_Load

End Class 'ViewerForm
